test_pull_requests

```diff
@@ -862,8 +1058,9 @@ popd
       repo_to_pull_request.each do |repo, pull_request|
         pull_request_comments = get_comments(pull_request['number'], repo)
         test_prefix_comment = get_comment_with_prefix(pull_request['number'], repo, settings['test_prefix'], pull_request_comments)
-        if !test_prefix_comment || !(test_prefix_comment['body'] =~ /^#{settings['test_prefix']} Waiting/)
-          being_queued_comment = "#{settings['test_prefix']} Waiting: Determining build queue position"
+        fields = extract_bot_comment_fields(test_prefix_comment['body'], settings) if test_prefix_comment
+        if !test_prefix_comment || !([:wait_in_queue, :wait_queue_pos, :wait_stable_build].include? fields[:state])
```

Comment:
So I'm still really unsure that we ever really want to do the check for all three of them. I posit that the original logic was incorrect and we just didn't notice or care since the cases where it was incorrect were rare enough. In this case, specifically:
 - we are processing a pull request
 - we know we are OK to trigger a new Jenkins job
 - if we were previously waiting for a downstream job, we need to update to say we're now in the build queue

Why do we need to update the comment if it currently says we're waiting to find our position in the queue, for example?